internal/controllers/s3userclaim/provisioner.go

```diff
@@ -311,3 +342,73 @@ func (r *Reconciler) assembleCephUserSecret(userName, secretName string) (*corev
 
 	return secret, nil
 }
+
+func (r *Reconciler) generateSubUserAccessMap(desiredSubUsers []string,
+	currentSubUsers []admin.SubuserSpec) map[string]string {
+	// Create a hashmap to move all spec and ceph subUsers to it
+	subUserFullIdAccessMap := make(map[string]string)
+
+	// Tag specSubUsers with "create"
+	for _, subUser := range desiredSubUsers {
+		cephSubUserFullId := generateSubUserFullId(r.cephUserFullId, subUser)
+		subUserFullIdAccessMap[cephSubUserFullId] = consts.SubUserTagCreate
+	}
+
+	// Add read-only subUser to subUsers to prevent if from removing
```

Comment:
Just a nit. `if` should be changed to `it`

Reply:
Done